requestor/ext/requestor-gwtjackson/src/main/java/io/reinert/requestor/ext/gwtjackson/SerializationModuleGenerator.java

```diff
@@ -0,0 +1,187 @@
+import io.reinert.requestor.ext.gwtjackson.processing.ProcessingLogger;
+
+public class SerializationModuleGenerator {
+
+    private final TypeElement moduleTypeElement;
+    private final TypeInfo moduleTypeInfo;
+    private final Set<JsonObjectSerdesAssembler> serdesAssemblers;
+    private SerializationModuleAssembler serializationModuleAssembler;
+    private boolean generated;
+
+    private final ProcessingLogger logger;
+
+    public SerializationModuleGenerator(Element element, ProcessingLogger logger) throws ProcessingException {
+        this.logger = logger;
+        checkElementIsInterface(element);
+
+        moduleTypeElement = (TypeElement) element;
+
+        checkInheritsFromSerializationModule(moduleTypeElement);
+
+        moduleTypeInfo = new TypeInfo(moduleTypeElement.getQualifiedName().toString());
+
+        serdesAssemblers = new LinkedHashSet<JsonObjectSerdesAssembler>();
+
+        processAnnotationValues(moduleTypeElement);
+
+        serializationModuleAssembler = new SerializationModuleAssembler(moduleTypeInfo, serdesAssemblers);
+    }
+
+    public Set<JsonObjectSerdesAssembler> getSerdesAssemblers() {
+        return serdesAssemblers;
+    }
+
+    public boolean isGenerated() {
+        return generated;
+    }
+
+    public void generate(Map<TypeInfo, JsonObjectSerdesGenerator> serdesGenerators, Filer filer) throws ProcessingException {
+        try {
+            mergeAndAssembleSerdes(serdesGenerators, filer);
+            serializationModuleAssembler.assemble();
+            JavaFile.builder(serializationModuleAssembler.packageName(), serializationModuleAssembler.spec())
+                    .build().writeTo(filer);
+        } catch (Exception e) {
+            throw new ProcessingException(moduleTypeElement, e,
+                    "Error while writing generated code of %s: %s", moduleTypeInfo.getSimpleName(), e.getMessage());
+        }
+        generated = true;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (!(o instanceof SerializationModuleGenerator))
+            return false;
+
+        final SerializationModuleGenerator that = (SerializationModuleGenerator) o;
+
+        return moduleTypeInfo.equals(that.moduleTypeInfo);
+    }
+
+    @Override
+    public int hashCode() {
+        return moduleTypeInfo.hashCode();
+    }
+
+    private void checkElementIsInterface(Element element) throws ProcessingException {
+        if (element.getKind() != ElementKind.INTERFACE) {
+            throw new ProcessingException(element, "Only interfaces can be annotated with @%s",
+                    JsonSerializationModule.class.getSimpleName());
+        }
+    }
+
+    private void checkInheritsFromSerializationModule(TypeElement element) throws ProcessingException {
+        for (TypeMirror superItf : element.getInterfaces()) {
+            if (SerializationModule.class.getName().equals(superItf.toString()))
```

Comment:
isValid is too generic. The method name already specifies what it is trying to achieve (or check).